src/commands/download_file.ts

```diff
@@ -0,0 +1,36 @@
+import { EID } from 'ardrive-core-js';
+import { resolve as resolvePath } from 'path';
+import { cliArDriveAnonymousFactory, cliArDriveFactory } from '../index';
+import { CLICommand, ParametersHelper } from '../CLICommand';
+import { CLIAction } from '../CLICommand/action';
+import {
+	DriveIdParameter,
+	DrivePrivacyParameters,
+	FileIdParameter,
+	LocalFilePathParameter
+} from '../parameter_declarations';
+
+new CLICommand({
+	name: 'download-file',
+	parameters: [FileIdParameter, LocalFilePathParameter, DriveIdParameter, ...DrivePrivacyParameters],
+	action: new CLIAction(async (options) => {
+		const parameters = new ParametersHelper(options);
+		const fileId = parameters.getRequiredParameterValue(FileIdParameter, EID);
+		const destFolderPath = resolvePath(parameters.getParameterValue(LocalFilePathParameter) || './');
+
+		if (await parameters.getIsPrivate()) {
+			const driveId = parameters.getRequiredParameterValue(DriveIdParameter);
+			const driveKey = await parameters.getDriveKey({ driveId: EID(driveId) });
+			const wallet = await parameters.getRequiredWallet();
+			const ardrive = cliArDriveFactory({
+				wallet,
+				feeMultiple: parameters.getOptionalBoostSetting()
+			});
+			await ardrive.downloadPrivateFile(fileId, destFolderPath, driveKey);
```

Comment:
What do you guys think of this behavior for -l:
• if the path is an existing folder path on disk, deposit the file in that folder based on its current name on ArDrive
• if the path is NOT an existing folder path on disk, assume the user wants to use it as the local file path, ignoring the file's name on ArDrive. To Discuss: Should we create any needed folder paths on the way to the file if they don't exist?

Reply:
That behavior might feel more natural to a user. 

Right now it seems to be only used as the folder path. If we keep this current behavior for now, I think we should at least use a new parameter with the name DestinationFolderPath or similar so that users don't confuse it with a file name parameter.

>  To Discuss: Should we create any needed folder paths on the way to the file if they don't exist?

I think yes. Creating folders for the user provides convenience and should be arbitrary with `fs` or similar

Reply:
> To Discuss: Should we create any needed folder paths on the way to the file if they don't exist?

I'd rather add a flag to force creating the directory tree the same way `mkdir -p` does. If the parent folder does not exist and no recursive flag provided, then throw the default `ENOENT`

Reply:
See the commit https://github.com/ardriveapp/ardrive-core-js/pull/91/commits/2412b211c28d722c3b99d392d644631773cdd872 at core for the localFilepathResolution algorithm

Reply:
> I'd rather add a flag to force creating the directory tree the same way mkdir -p does. If the parent folder does not exist and no recursive flag provided, then throw the default ENOENT

Let's make a decision based on the ease of use of the following permutations:
1) -l path is a folder on the local file system that exists
    • Result: CLI uses folder as destPath and provides an undefined filename, leading to the ArDrive filename being used in the full path.
2) -l path is a file on the local file system that exists
    • Result: CLI uses the parent folder of the file as destPath and provides the filename. ArDrive filename is ignored.
3) -l path is a nonexistent entity on the local file system, but whose parent folder exists
    • Result: CLI uses the parent folder of the entity as destPath and provides the entity name as a filename. ArDrive filename is ignored. (DOES THIS SEEM CORRECT?!?)
3) -l path is a nonexistent entity on the local file system whose parent folder does NOT exist
    • Result: ERROR

Let's see how a flag like `mkdir`'s `-p` would change these cases:
1) No change
2) No change
3) No change, DEPENDING on whether we think the behavior described above is desirable
4) The path to the parent folder of the entity is created. The entity name is used. ArDrive name is ignored.

@agsuy should weigh in here.

Reply:
Ideas for the flag, if we ever use it:
--create-paths
--parents
--create-intermediates alias: -i